### PR TITLE
ENG-19118: return roles as list instead of array

### PR DESCRIPTION
**Pull request by rdykiel**

### CODE REVIEW

File: src/frontend/org/voltdb/AuthSystem.java [hunk 1]
```diff
@@ -283,11 +283,9 @@ public boolean hasPermission(Permission... perms) {
...
          * @return group name array
          */
-        public final String[] getGroupNames() {
-            String[] groupNames = new String[m_groups.size()];
-            for (int i = 0; i < m_groups.size(); ++i) {
-                groupNames[i] = m_groups.get(i).m_name;
-            }
+        public final List<String> getGroupNames() {
+            List<String> groupNames = new ArrayList<>(m_groups.size());
+            m_groups.forEach(g -> groupNames.add(g.m_name));
```
Comment (n3world) [inline src/frontend/org/voltdb/AuthSystem.java]: Why are you doing this just return m_groups it is immutable? Or atl east just use the construct which takes a Collection `new ArrayList<>(m_groups);`

File: src/frontend/org/voltdb/AuthSystem.java [hunk 2]
```diff
@@ -670,13 +668,13 @@ public AuthUser getUser(String name) {
         return m_users.get(name);
     }
 
-    public String[] getGroupNamesForUser(String userName) {
+    public List<String> getGroupNamesForUser(String userName) {
         if (userName == null) {
-            return new String[] {};
+            return new ArrayList<>();
```
Comment (n3world) [inline src/frontend/org/voltdb/AuthSystem.java]: Don't construct a list return either Collections.emptyList() or ImmutableList.of();

File: src/frontend/org/voltdb/AuthSystem.java [hunk 3]
```diff
@@ -670,13 +668,13 @@ public AuthUser getUser(String name) {
...
         }
         AuthUser user = getUser(userName);
         if (user == null) {
-            return new String[] {};
+            return new ArrayList<>();
```
Comment (n3world) [inline src/frontend/org/voltdb/AuthSystem.java]: Same

File: src/frontend/org/voltdb/AuthSystem.java (hunk 1 above)
Comment (rdykiel) [reply src/frontend/org/voltdb/AuthSystem.java]: Because m_goups is a list of GROUPS and we want the list of group NAMES!!

**Pull request merged**